data/campaigns/Eastern_Invasion/scenarios/04b_Ill_Humours.cfg

```diff
@@ -0,0 +1,1064 @@
+        {MOVE_UNIT (id=Nur al-Nuh)   2 25}  {KILL (id=Nur al-Nuh)}
+        {MOVE_UNIT (id=Muma)         2 25}  {KILL (id=Muma)}
+        [redraw]
+            clear_shroud=yes
+        [/redraw]
+        [message]
+            speaker=Gweddry
+            message=_ "The dunefolk are leaving and without their aid we have no chance of even finding the undead capital. We must defeat Mal-Skraat and proceed westward."
+        [/message]
+        [show_objectives]
+        [/show_objectives]
+    [/event]
+
+    #------------------
+    # MAL-SKRAAT DIES
+    #------------------
+    [event]
+        name=last breath
+        [filter]
+            id=Mal-Skraat
+        [/filter]
+
+        [message]
+            speaker=Mal-Skraat
+            message=_ "No, it can’t end like this! My sister has died unavenged..."
+        [/message]
+        {KILL id=Mal-Skraat ANIMATE=yes}
+        [message]
+            speaker=Dacyn
+            message= _ "Good, now we should be able to reach what’s left of the northern outpost."
+        [/message]
+        [if] {VARIABLE_CONDITIONAL herbs_needed greater_than 0}
+            [then]
+                {MODIFY_UNIT (id=Hahid al-Ali) side 4}
+                {MODIFY_UNIT (id=Safyam) side 4}
+                [message]
+                    speaker=Hahid al-Ali
+                    message= _ "Thanks for nothing! What happened to helping me gather my herbs?"
+                [/message]
+            [/then]
+        [/if]
+
+        {VARIABLE skraat_dead yes}
+        {VARIABLE fork_04c no}
+        [endlevel]
+            result=victory
+            bonus=yes
+            next_scenario=05_Northern_Outpost
+            {NEW_GOLD_CARRYOVER 40}
+        [/endlevel]
+    [/event]
+
+    [event]
+        name=victory
+        # because for some reason this doesn't get cleared when the objects are removed,
+        # even back when I had it as part of the object and not a separate effect
+        [modify_unit]
+            [filter]
+                side=1
+            [/filter]
+
+            [status]
+                unpoisonable=no
+            [/status]
+        [/modify_unit]
+
+        {CLEAR_VARIABLE gold,gold_cost,herbs_needed}
+        {CLEAR_VARIABLE loc1_x,loc1_y}
+        {CLEAR_VARIABLE loc2_x,loc2_y}
+        {CLEAR_VARIABLE loc3_x,loc3_y}
+        {CLEAR_VARIABLE loc4_x,loc4_y}
+        {CLEAR_VARIABLE loc5_x,loc5_y}
+        {CLEAR_VARIABLE loc6_x,loc6_y}
+        {CLEAR_VARIABLE loc7_x,loc7_y}
+        {CLEAR_VARIABLE loc8_x,loc8_y}
+        {CLEAR_VARIABLE loc9_x,loc9_y}
+        {CLEAR_VARIABLE loc10_x,loc10_y}
+        {CLEAR_VARIABLE loc11_x,loc11_y}
+        {CLEAR_VARIABLE loc12_x,loc12_y}
+        {CLEAR_VARIABLE loc13_x,loc13_y}
+        {CLEAR_VARIABLE loc14_x,loc14_y}
+        {CLEAR_VARIABLE loc15_x,loc15_y}
+        {CLEAR_VARIABLE loc16_x,loc16_y}
+        {CLEAR_VARIABLE loc17_x,loc17_y}
+        {CLEAR_VARIABLE loc18_x,loc18_y}
+        {CLEAR_VARIABLE loc19_x,loc19_y}
```

Comment:
The variables are set further up, in a `[random_placement]`, and are used so that each herb can have its own event without overwriting each-others' variables.
```
    [random_placement]
        variable=location
        num_items=1
        [filter_location]
            x,y={X},{Y}
            terrain=Ss,Sm,Sm^Qhu
        [/filter_location]
        [command]
            {VARIABLE loc{I}_x $location.x}
            {VARIABLE loc{I}_y $location.y}
            {PLACE_HERB $loc{I}_x $loc{I}_y}
        [/command]
    [/random_placement]
```
If this can be done with an array instead that's a far better solution.

Reply:
I think this should work:

```
[command]
    {PLACE_HERB $location.x $location.y}
    [set_variables]
        name=herb_locations
        mode=append
        to_variable=location
        # Or if to_variable doesn't work: [value] x,y=$location.x,$location.y
[/command]
```

Or another approach:

```
[command]
    {PLACE_HERB $location.x $location.y}
    {VARIABLE herb_locations[{I}].x $location.x}
    {VARIABLE herb_locations[{I}].y $location.y}
[/command]
```